app/Http/Controllers/CommentController.php

```diff
@@ -15,117 +18,129 @@ public function index(Post $post)
-    }
+        $commentData = [
+            'temat' => $request->input('temat'),
+            'user_id' => $user_id,
+        ];
 
-    if ($request->hasFile('image')) {
-        if ($comment->imageExists()) {
+        if ($request->has('remove_image') && $comment->imageExists()) {
             Storage::disk('public')->delete($comment->image_path);
+            $commentData['image_path'] = null;
         }
 
-        $imagePath = $request->file('image')->store('comments', 'public');
-        $commentData['image_path'] = $imagePath;
-    }
+        if ($request->hasFile('image')) {
+            if ($comment->imageExists()) {
+                Storage::disk('public')->delete($comment->image_path);
+            }
 
-    $post = $comment->post;
+            $commentData['image_path'] = $request->file('image')->store('comments', 'public');
+        }
 
-    $comment->update($commentData);
+        $post = $comment->post;
 
-    return redirect()->route('comments.index', $post->id)->with('success', 'Komentarz został zaktualizowany.');
-}
+        $comment->update($commentData);
 
+        return redirect()->route('comments.index', $post->id)->with('success', ' Comment update.');
+    }
 
-public function destroy(Comment $comment): JsonResponse
-{
-    try {
-        if ($comment->imageExists()) {
-            Storage::disk('public')->delete($comment->image_path);
-        }
+    public function destroy(Post $post, Comment $comment): JsonResponse
+    {
 
-        $comment->likes()->delete();
+        $user = Auth::user();
 
-        $comment->delete();
+        try {
+            if ($comment->imageExists()) {
+                Storage::disk('public')->delete($comment->image_path);
+            }
 
-        Session::flash('success', 'Udało się usunąć comment.');
+            $comment->likes()->delete();
 
-        return response()->json([
-            'status' => 'success'
-        ]);
-    } catch (\Exception $e) {
-        \Log::error($e);
+            $comment->delete();
+
+            Session::flash('success', 'Comment destroy.');
 
-        if ($comment->imageExists()) {
-            \Log::info('Error occurred. Not deleting image from storage.');
+            return response()->json([
+                'status' => 'success',
+            ]);
+        } catch (\Exception $e) {
+            \Log::error($e);
+
+            return response()->json([
+                'status' => 'error',
+                'message' => 'Error!',
+            ])->setStatusCode(500);
         }
 
-        return response()->json([
-            'status' => 'error',
-            'message' => 'Wystąpił błąd!'
-        ])->setStatusCode(500);
     }
-}
 
-public function like($commentId)
-{
-    $comment = Comment::findOrFail($commentId);
-    $user = Auth::user();
+    public function like(Post $post, Comment $comment): JsonResponse
+    {
+        if (! $comment) {
+            return response()->json(['status' => 'error', 'message' => 'Comment not found'], 404);
+        }
```

Comment:
Jak masz rzutowanie na Comment to nie potrzebujesz tego ifa, bo jak nie znajdzie to samo wyjebie 404